### PR TITLE
docs.md: document [noinit] attribute with example

### PR DESCRIPTION
**Pull request by hungrybluedev**
- Documents the `[noinit]` attribute for structs.
- Closes https://github.com/vlang/v/issues/15655

### CODE REVIEW

File: doc/docs.md [hunk 1]
```diff
@@ -2139,6 +2139,52 @@ assert book.author.name == 'Samantha Black'
...
+		data: data
+	}
+}
+```
+
+Note that `new_information` is a _factory_ function. Now when we want to make use of this struct
+outside the module:
+
+```v oksyntax
+// ... Imports and stuff ...
```
Comment (Larpon) [inline doc/docs.md:2173]: Maybe just:
```suggestion
import sample
```

File: doc/docs.md [hunk 2]
```diff
@@ -2139,6 +2139,52 @@ assert book.author.name == 'Samantha Black'
...
+	if data.len == 0 || data.len > 100 {
+		return error('data must be between 1 and 100 characters')
+	}
+	return Information{
+		data: data
+	}
+}
+```
+
+Note that `new_information` is a _factory_ function. Now when we want to make use of this struct
```
Comment (spytheman) [inline doc/docs.md:2169]:
```suggestion
Note that `new_information` is a _factory_ function. Now when we want to use this struct
```

File: doc/docs.md [hunk 3]
```diff
@@ -2139,6 +2139,52 @@ assert book.author.name == 'Samantha Black'
...
 
+### `[noinit]` structs
+
+V supports `[noinit]` structs, which are structs that cannot be initialised outside the module
+they are defined in. They are either meant to be used internally or they can be used externally
+through _factory functions_.
+
+For an example, consider the following source in a directory `sample`:
+
+```v oksyntax
```
Comment (spytheman) [inline doc/docs.md:2150]:
```suggestion
```v ignore
```

**Comment by hungrybluedev** [comment]
I tried keeping the `import sample` in but it doesn't compile under the check-md step. That's why I commented it.

**Comment by spytheman** [comment]
> I tried keeping the `import sample` in but it doesn't compile under the check-md step. That's why I commented it.

I'll implement support for a `v okfmt` markdown tag.

**Pull request merged**